### ISSUE TITLE
Build failure with newest zig master due to new alignForward definition

### ISSUE THREAD

**Issue by DilithiumNitrate** [zigtools/zls#1244]
### Zig Version

0.11.0-dev.3681+d41111d7e

### Zig Language Server Version

f6da47fafdbc89ab628613fea52897c720b96564

### Steps to Reproduce

Build zls with the specified version of Zig, with `zig build -Doptimize=ReleaseSafe`.

### Expected Behavior

I'd expect the build to succeed. (Not much to say here.)

### Actual Behavior

The build fails and following error message is printed:
```
Build Summary: 1/4 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install zls transitive failure
   └─ zig build-exe zls ReleaseSafe native 1 errors
src/analyser/encoding.zig:196:27: error: expected 3 argument(s), found 2
    const offset = std.mem.alignForward(unaligned, alignment) - unaligned;
                   ~~~~~~~^~~~~~~~~~~~~
/home/strife/.local/opt/zig/lib/std/mem.zig:4216:5: note: function declared here
pub fn alignForward(comptime T: type, addr: T, alignment: T) T {
~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    decode__anon_70774: src/analyser/encoding.zig:134:35
    decode__anon_61744: src/analyser/encoding.zig:163:54
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```
The origin of this error can be directly attributed to [commit d41111d7ef531f6f55a19c56205d6d2f1134c224 in Zig, `mem: rename align*Generic to mem.align*`](https://github.com/ziglang/zig/commit/d41111d7ef531f6f55a19c56205d6d2f1134c224), committed less than a day ago. The obvious course of action to fix this (tested, works; should I just make a pull request?) would be to change the call to `std.mem.alignForward` to pass the alignment's type – in this case `usize` – as per the new definition, formerly known as `std.mem.alignForwardGeneric`.

**Comment by leecannon** [comment]
Zig had a breaking change yesterday [here](https://github.com/ziglang/zig/pull/15992).
You will need to update zig to compile zls master.

**Comment by leecannon** [comment]
Sorry about that, this is is actually another breaking change i'm looking at fixing it now.

**Comment by leecannon** [comment]
This will be fixed by #1246
